src/main/java/org/ilastik/ilastik4ij/ui/IlastikImportModel.java

```
@@ -0,0 +1,163 @@
+package org.ilastik.ilastik4ij.ui;
+
+import org.ilastik.ilastik4ij.hdf5.HDF5DatasetEntryProvider;
+import org.ilastik.ilastik4ij.hdf5.HDF5DatasetEntryProvider.DatasetEntry;
+import org.scijava.log.LogService;
+
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+import java.util.Vector;
+
+class IlastikImportModel {
+    public static final String PROPERTY_PATH = "path";
+    public static final String PROPERTY_DATASET_IDX = "datasetIdx";
+    public static final String PROPERTY_AXIS_TAGS = "axisTags";
+
+    private String path = "";
+    private int datasetIdx = -1;
+    private String axisTags = "";
+
+    private boolean isPathValid = false;
+
+    private Vector<HDF5DatasetEntryProvider.DatasetEntry> availableDatasets = new Vector<>();
+    private final LogService logService;
+    private final PropertyChangeSupport propertyChangeSupport;
+    private final HDF5DatasetEntryProvider entryProvider;
+
+    public IlastikImportModel(LogService logService) {
+        this.logService = logService;
+        propertyChangeSupport = new PropertyChangeSupport(this);
+        entryProvider = new HDF5DatasetEntryProvider(logService);
+    }
+
+    public int getDatasetIdx() {
+        return this.datasetIdx;
+    }
+
+    public boolean isValid() {
+        return this.isPathValid && this.isDatasetIdxValid() && this.isAxisTagsValid();
+    }
+
+    public String getPath() {
+        return path;
+    }
+
+    public boolean isPathValid() {
+        return this.isPathValid;
+    }
+
+    public void setPath(String path) {
+        if (this.path.equals(path)) {
+            return;
+        }
+
+        this.isPathValid = true;
+        String oldPath = this.path;
+        this.path = path;
+
+        try {
+            availableDatasets = entryProvider.findAvailableDatasets(path);
+        } catch (Exception e) {
```

more a question: is it common in java to catch all Exceptions as opposed to specific exceptions one expects?